src/ASM/ASMs2D.h

```diff
@@ -529,6 +529,7 @@ class ASMs2D : public ASMstruct, public ASM2D
   //! \param[out] XC Coordinates of the element corners
   void getElementCorners(int i1, int i2, std::vector<Vec3>& XC) const;
 
+  using ASMbase::generateThreadGroups;
   //! \brief Generates element groups for multi-threading of interior integrals.
```

Comment:
ASMbase expose https://github.com/OPM/IFEM/blob/master/src/ASM/ASMbase.h#L375 and https://github.com/OPM/IFEM/blob/master/src/ASM/ASMbase.h#L377

ASMs2D only implements https://github.com/OPM/IFEM/blob/master/src/ASM/ASMs2D.h#L536

this hides https://github.com/OPM/IFEM/blob/master/src/ASM/ASMbase.h#L377 from its children, which is what the warning is all about.

there is no warning in the children, because they only see https://github.com/OPM/IFEM/blob/master/src/ASM/ASMs2D.h#L536 and they thus implement all overloads inherited from their children.